lib/ansible/modules/cloud/amazon/cloudfront_facts.py

```diff
@@ -270,68 +286,153 @@ def get_streaming_distribution_config(self, distribution_id):
+            return distribution_list
+        except Exception as e:
+            self.module.fail_json(msg="Error generating summary of distributions - " + str(e), exception=traceback.format_exc(e))
+
+    def get_etag_from_distribution_id(self, distribution_id, streaming):
+        distribution = {}
+        if not streaming:
+            distribution = self.get_distribution(distribution_id)
+        else:
+            distribution = self.get_streaming_distribution(distribution_id)
+        return distribution['ETag']
+
+    def get_list_of_invalidation_ids_from_distribution_id(self, distribution_id):
+        try:
+            invalidation_ids = []
+            invalidations = self.list_invalidations(distribution_id)
+            for invalidation in invalidations:
+                invalidation_ids.append(invalidation['Id'])
+            return invalidation_ids
+        except Exception as e:
+            self.module.fail_json(msg="Error getting list of invalidation ids - " + str(e), exception=traceback.format_exc(e))
 
     def get_distribution_id_from_domain_name(self, domain_name):
         try:
             distribution_id = ""
             distributions = self.list_distributions(False)
+            distributions += self.list_streaming_distributions(False)
             for dist in distributions:
-                for alias in dist['Aliases']['Items']:
-                    if str(alias).lower() == domain_name.lower():
-                        distribution_id = str(dist['Id'])
-                        break
+                if 'Items' in dist['Aliases']:
+                    for alias in dist['Aliases']['Items']:
+                        if str(alias).lower() == domain_name.lower():
+                            distribution_id = dist['Id']
+                            break
             return distribution_id
         except Exception as e:
-            self.module.fail_json(msg="Error getting distribution id from domain name = " + str(e), exception=traceback.format_exc(e))
+            self.module.fail_json(msg="Error getting distribution id from domain name - " + str(e), exception=traceback.format_exc(e))
 
     def get_aliases_from_distribution_id(self, distribution_id):
         aliases = []
         try:
             distributions = self.list_distributions(False)
             for dist in distributions:
-                if dist['Id'] == distribution_id:
+                if dist['Id'] == distribution_id and 'Items' in dist['Aliases']:
                     for alias in dist['Aliases']['Items']:
                         aliases.append(alias)
                     break
             return aliases
         except Exception as e:
-            self.module.fail_json(msg="Error getting list of aliases from distribution_id = " + str(e), exception=traceback.format_exc(e))
+            self.module.fail_json(msg="Error getting list of aliases from distribution_id - " + str(e), exception=traceback.format_exc(e))
```

Comment:
Worth wrapping all these long fail_json calls at the comma before exception - otherwise it goes off the side at PR (I know there are lots of lines that are already too long but let's not add more)